src/Speckle.Sdk/Models/DynamicBase.cs

```diff
@@ -153,78 +151,54 @@ public bool IsPropNameValid(string name, out string reason)
     return true;
   }
 
-  private static void PopulatePropInfoCache(Type type)
-  {
-    if (!s_propInfoCache.ContainsKey(type))
-    {
-      s_propInfoCache[type] = type.GetProperties(BindingFlags.Instance | BindingFlags.Public)
-        .Where(p => !p.IsDefined(typeof(IgnoreTheItemAttribute), true))
-        .ToList();
-    }
-  }
-
   /// <summary>
   /// Gets all of the property names on this class, dynamic or not.
   /// </summary> <returns></returns>
   [Obsolete("Use `GetMembers(DynamicBaseMemberType.All).Keys` instead")]
   public override IEnumerable<string> GetDynamicMemberNames()
   {
-    PopulatePropInfoCache(GetType());
-    var pinfos = s_propInfoCache[GetType()];
+    var pinfos = TypeLoader.GetBaseProperties(GetType());
 
-    var names = new List<string>(_properties.Count + pinfos.Count);
     foreach (var pinfo in pinfos)
     {
-      names.Add(pinfo.Name);
+      yield return pinfo.Name;
     }
 
     foreach (var kvp in _properties)
     {
-      names.Add(kvp.Key);
+      yield return kvp.Key;
     }
-
-    return names;
   }
 
   public static IEnumerable<string> GetInstanceMembersNames(Type t)
   {
-    PopulatePropInfoCache(t);
-    var pinfos = s_propInfoCache[t];
+    var pinfos = TypeLoader.GetBaseProperties(t);
 
-    var names = new List<string>(pinfos.Count);
     foreach (var pinfo in pinfos)
     {
-      names.Add(pinfo.Name);
+      yield return pinfo.Name;
     }
-
-    return names;
   }
 
   /// <summary>
   /// Gets the defined (typed) properties of this object.
   /// </summary>
   /// <returns></returns>
-  public IReadOnlyList<PropertyInfo> GetInstanceMembers()
+  public IEnumerable<PropertyInfo> GetInstanceMembers()
   {
     return GetInstanceMembers(GetType());
   }
 
-  public static IReadOnlyList<PropertyInfo> GetInstanceMembers(Type t)
+  public static IEnumerable<PropertyInfo> GetInstanceMembers(Type t)
```

Comment:
Unsure about this change, 
returning a fully enumerated list means consumers know the size (so they can pre-size their collections appropriately). we ourselves use this in [here](https://github.com/specklesystems/speckle-sharp-sdk/blob/dc4da4907851ef581083926da83f24be7b37f1eb/src/Speckle.Sdk/Serialisation/BaseObjectSerializerV2.cs#L500-L501)

Imo, I'd actually prefer we return correctly pre-sized `IReadOnlyList` everywhere in this class... (maybe except for the overrides)

Reply:
I'd rather not do allocations unless needed.   Only one place needed it.